File: app/src/main/java/co/lnu/elevatorapp/elevator_simultion/ElevatorSimulationPresenter.java
package co.lnu.elevatorapp.elevator_simultion;

import co.lnu.elevatorapp.Elevator;
import co.lnu.elevatorapp.Floor;

import java.util.List;

public interface ElevatorSimulationPresenter {
    List<Floor> getListOfFloors();

    void setFloorHeight(int height);

    float moveToFloorYDelta(int intendedFloor);

    int getFloorHeight();

    void onViewCreated(int floorNumber, int elevatorNumber);

    List<Elevator> getListOfElevators();
}

File: app/src/main/java/co/lnu/elevatorapp/elevator_simultion/ElevatorSimulationPresenter.kt
package co.lnu.elevatorapp.elevator_simultion

import co.lnu.elevatorapp.Floor

interface ElevatorSimulationPresenter
{

    fun getListOfFloors(): List<Floor>

    fun setFloorHeight(height: Int)

    fun moveToFloorYDelta(intendedFloor: Int): Float

}